### PR TITLE
MAINT: Chaining exceptions in numpy/core/_internal.py

### PR DESCRIPTION
**Pull request by abhinav-thimma**
Changes related to #15986
- added chaining for ValueError, TypeError, AttributeError
- updated % formatted strings to f strings

### CODE REVIEW

File: numpy/core/_internal.py [hunk 1]
```diff
@@ -371,11 +372,11 @@ def _newnames(datatype, order):
...
             try:
                 nameslist.remove(name)
-            except ValueError:
+            except ValueError as e:
                 if name in seen:
-                    raise ValueError("duplicate field name: %s" % (name,))
+                    raise ValueError(f"duplicate field name: {name}") from e
                 else:
-                    raise ValueError("unknown field name: %s" % (name,))
+                    raise ValueError(f"unknown field name: {name}") from e
```
Comment (eric-wieser) [inline numpy/core/_internal.py:379]: These two should be `from None`

File: numpy/core/_internal.py [hunk 2]
```diff
@@ -161,9 +161,10 @@ def _commastring(astr):
         mo = format_re.match(astr, pos=startindex)
         try:
             (order1, repeats, order2, dtype) = mo.groups()
-        except (TypeError, AttributeError):
-            raise ValueError('format number %d of "%s" is not recognized' %
-                                            (len(result)+1, astr))
+        except (TypeError, AttributeError) as e:
+            raise ValueError(
+                f'format number {len(result)+1} of "{astr}" is not recognized'
+                ) from e
```
Comment (eric-wieser) [inline numpy/core/_internal.py:167]: Also should be `from None`, the error here isn't interesting

Comment (abhinav-thimma) [reply numpy/core/_internal.py:167]: @eric-wieser  I will make the necessary corrections.

**Comment by seberg** [comment]
LGTM, thanks @abhinav-thimma and Eric.

**Pull request merged**